### ISSUE TITLE
Minor issue with reproducible builds

### ISSUE THREAD

**Issue by IzzySoft** [SlaVcE14/JsonList#44]
At IzzyOnDroid we support [Reproducible Builds](https://reproducible-builds.org/) (see: [Reproducible Builds, special client support and more at IzzyOnDroid](https://android.izzysoft.de/articles/named/iod-rbs-mirrors-clients)). While your app builds fine, there's an issue with the PNGs not being reproducible. Could it be you're using [some image optimization](https://gist.github.com/obfusk/f0460dcdb21396ba3d2cc9db56bb40ae#differing-png-files-easy-to-fix) (PNGCrunch & Co) at build time? Those are mostly non-deterministic, and thus better applied before checking in the images. It could also be [PNGs generated from vector drawables](https://f-droid.org/en/docs/Reproducible_Builds/#pngs-generated-from-vector-drawables). Both cases are easy to fix with a short addition to your `build.gradle`.

For reference, [this is the commit](https://github.com/obfusk/rbtlog/commit/4d0bc85528ee150a08ade29ed92d35f63752ee65) in one of our builders marking this.

We'd appreciate if you could help making your build reproducible. We've also prepared some [hints on reproducible builds](https://gitlab.com/IzzyOnDroid/repo/-/wikis/Reproducible-Builds) for that.

Looking forward to your reply!

cc @obfusk

**Comment by SlaVcE14** [comment]
I think it's fixed now.
Do I need to make a new release?

**Comment by licaon-kter** [comment]
I tried to build 52527022b7d9ac3dc9da0aeda008aa392fef7111 and use the release/ APK: https://github.com/SlaVcE14/JsonList/tree/52527022b7d9ac3dc9da0aeda008aa392fef7111/app/release

but
```
==== detail begin ====
verification of APK with copied signature failed
Comparing reference APK to APK with copied signature...
Unexpected diff output:
diff -r /tmp/tmps89bkz19/unsigned_binaries_com.sjapps.jsonlist_20.binary/content/META-INF/version-control-info.textproto /tmp/tmps89bkz19/_tmp_tmps89bkz19_sigcp_com.sjapps.jsonlist_20/content/META-INF/version-control-info.textproto
diff -r /tmp/tmps89bkz19/unsigned_binaries_com.sjapps.jsonlist_20.binary/content/META-INF/version-control-info.textproto /tmp/tmps89bkz19/_tmp_tmps89bkz19_sigcp_com.sjapps.jsonlist_20/content/META-INF/version-control-info.textproto
4c4
<   revision: "7f90bf4356d36107b2ea3144e4f2902494f18246"
---
>   revision: "52527022b7d9ac3dc9da0aeda008aa392fef7111"
==== detail end ====
```
looks like the attached APK was build from a tree at 7f90bf4356d36107b2ea3144e4f2902494f18246 maybe with uncommited code? @SlaVcE14

**Comment by SlaVcE14** [comment]
Yes, the attached APK and the APK in the [releases](https://github.com/SlaVcE14/JsonList/tree/52527022b7d9ac3dc9da0aeda008aa392fef7111/app/release) in the last commit si not the same. Do I only need to change the attached APK in the [1.5 release](https://github.com/SlaVcE14/JsonList/releases/tag/v1.5) or remove the 1.5 release and make a new release? @licaon-kter

**Comment by licaon-kter** [comment]
not sure how that affects @IzzySoft now, you could attach it here for us to test, rename it to .APK.ZIP (do not archive it!)

**Comment by SlaVcE14** [comment]
[JsonList_1.5.apk.zip](https://github.com/user-attachments/files/17292389/JsonList_1.5.apk.zip)

**Comment by licaon-kter** [comment]
@SlaVcE14 that attached APK is still built from 7f90bf4356d36107b2ea3144e4f2902494f1824 

please clone a clean tree, checkout 52527022b7d9ac3dc9da0aeda008aa392fef7111 and build it

**Comment by SlaVcE14** [comment]
[app-release - 1.5.apk.zip](https://github.com/user-attachments/files/17292682/app-release.-.1.5.apk.zip)

**Comment by licaon-kter** [comment]
@SlaVcE14 :+1: :tada: that one if fine

**Comment by IzzySoft** [comment]
> Do I need to make a new release?

Yes please. Except for very rare occasions (usually security-related), one should never replace existing releases which have already been distributed. And for RB to work, the APK must match the commit the tag points to anyway – see the first golden rule in our hints above.

I'm sure LK wants it the same way for F-Droid. The attachment here is fine for testing (we do the same at IzzyOnDroid), but for distribution it must be a release.

Thanks! Also to @licaon-kter for testing :wink:
